src/DotLiquid.Tests/DropTests.cs

```diff
@@ -353,6 +370,33 @@ public void TestAccessContextFromDrop()
             Assert.AreEqual("123", Template.Parse("{% for a in dummy %}{{ context.loop_pos }}{% endfor %}").Render(Hash.FromAnonymousObject(new { context = new ContextDrop(), dummy = new[] { 1, 2, 3 } })));
         }
 
+        [Test]
+        public void TestCancelBlockingOperationDrop()
+        {
+            var cancellationToken = new CancellationToken(true);

```

Comment:
Use a cancellation token source to get a cancellation token. see for instance https://docs.microsoft.com/en-us/dotnet/api/system.threading.cancellationtoken?view=net-5.0#examples

Reply:
I've switched to using a `CancellationTokenSource` now.

Reply:
Still using the Slim, and cancellation is before the render. I think an ideal way to test this function is to have a context variable with number of seconds to sleep and see the framework cancel if the sleep > timeout. The same template will pass/fail depending on the number of seconds to sleep in the drop.

Reply:
By the way, Context.CheckTimeout() is public, checks the cancellation token, and can already be accessed from the drop. Not seeing what making cancellationToken public is adding - as that is NOT checking timeout. Example:
```c#
        internal class CancellationDrop : Drop
        {
            public string Halt
            {
                get
                {
                    var timeToWait = (int)Context["sleep_time"];
                    while (timeToWait > 0)
                    {
                        Context.CheckTimeout();
                        var sleepTime = Math.Min(timeToWait, 50);
                        System.Threading.Thread.Sleep(sleepTime);
                        timeToWait -= sleepTime;
                    }

                    return "Here";
                }
            }
        }
```
The only missing piece is NATIVE evaluation of a drop property with a timeout. That would allow for a getter:
```c#
                    System.Threading.Thread.Sleep((int)Context["sleep_time"]);
                    return "Here";
```